torch/serialization.py

```diff
@@ -161,6 +161,25 @@ def _should_read_directly(f):
         return False
 
 
+def _can_tell_and_seek(f):
+
+    def raise_load_err_msg(patterns, e):
+        for p in patterns:
+            if p in str(e):
+                msg = (str(e) + ". You can only torch.load from a file that is seekable." +
+                                " Please pre-load the data into a buffer like io.BytesIO and" +
+                                " try to load from it instead.")
+                raise type(e)(msg)
+            else:
+                raise e
```

Comment:
Let's say the exception is `RuntimeError("method not found: tell")`. `patterns = ["seek", "tell"]` here. 

The first time we go into the for loop, `p in str(e)` is going to resolve to False, so we're going to `raise e`. So `RuntimeError("method not found: tell")` is raised instead of the desired `RuntimeError("method not found: tell. You can only torch.load from a file that is seekable...")` that you want to raise

Reply:
Good catch! I will change it accordingly.